### ISSUE TITLE
Octopus duplicate parent error

### ISSUE THREAD

**Issue by danoliv** [lesfurets/git-octopus#32]
Hi, we have found a problem with git octopus in our environment, when we have one branch that is aligned with master and at least one that is not. 

In this case **git octopus return a success but the merge is not done** and a "duplicate parent error" appears.

Here how to reproduce the problem:

- first we prepare the repository
```
mkdir gitOctopusMergeIssue
cd gitOctopusMergeIssue/
git init
echo "line1" > testFile
git add testFile
git commit -m"Add first file"
git checkout -b feature1
git checkout -b feature2
echo "line2" >> testFile
git commit -am"Add line2"
```
- then we launch git octopus from master 
```
git checkout master
git octopus feature1 feature2
Branches beeing merged :
	refs/heads/feature1
	refs/heads/feature2
-----------------------------------------------------------
Already up-to-date with refs/heads/feature1
Fast-forwarding to: refs/heads/feature2
error: duplicate parent 78924a5b99166dd6c4f73ec6f95a9232c4c752f6 ignored
-----------------------------------------------------------
OCTOPUS SUCCESS
```
- we see that the merge has not been done even if git octopus return a SUCCESS
```
cat testFile 
line1
```
-  "line2" from branch feature2 is missing

**Comment by apflieger** [comment]
I wrote a test on `duplicate_parent_error` branch to reproduce this usecase. I guess you are running on the lastest release, note that I'm using the go rewrite version here.
It seems that the error is not reproduced this way. The output of git-octopus is
```
2 branches (I)ncluded (2 matching, 0 (E)xcluded):
I  refs/heads/feature1
I  refs/heads/feature2

Already up-to-date with refs/heads/feature1
Fast-forwarding to: refs/heads/feature2
```
See here https://travis-ci.org/lesfurets/git-octopus/builds/218447494
So we get the expected result

**Comment by danoliv** [comment]
We are running version 1.4 that is the last stable release, we are waiting for a stable 2.0 in order to package it properly (rpm) and then use it in production. 
Do you think that the current 2.0-beta1 is production ready? 
If no, do you plan to release a stable 2.0 soon?
Thanks

**Comment by apflieger** [comment]
First thing is that I doubt this behavior is related to the version. It's just that the reprodution steps are not reproducing the problem :)
I miss a bit of testing on 2.0 and there are some features that are not implemented yet like the conflict resolution thing.

**Comment by danoliv** [comment]
I thought it could be due to the git version but we reproduced the problem with different version of git 2.9.3 (centos 6 and fedora 25), 2.11.0 (centos 6), 1.8.3.1 (centos 7).
We are going to test the go rewrite in order to see if it fixes our problem.

**Comment by danoliv** [comment]
We tested with the go version, so the problem is already fixed on the 2.0 like you saw in your test.
If you want to reproduce it try with the version 1.4.
